Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM fedora:25 as builder
+
+RUN dnf install -y \
+    git-core \
+    gcc \
+    gcc-c++ \
+    boost-devel \
+    cmake \
+    autoconf \
+    automake \
+    libtool \
+    mariadb-devel \
+    file \
+    hostname \
+    mariadb-server
+
+RUN git clone --branch master --single-branch --depth 1 https://github.com/anton-povarov/meow /_src/meow
+RUN git clone --branch master --single-branch --depth 1 https://github.com/nanomsg/nanomsg /_src/nanomsg
```

Comment:
@anton-povarov I would recommend lock `nanomsg` to a specific release for example 
https://github.com/nanomsg/nanomsg/releases/tag/1.1.5 and download & extract tar.gz using `ADD`.